src/yaml.cr

```diff
@@ -147,4 +148,28 @@ module YAML
   def self.dump(object, io : IO)
     object.to_yaml(io)
   end
+
+  # Determines if the scaler is reserved
+  def self.reserved_scalar(value : String)
```

Comment:
@ysbaddaden something like this. Although I havent tested it yet, it would fail because not all subclasses support the required arguments for new.

```crystal
def Class.new(pull : YAML::PullParser)
  return pull.read_scalar if pull.style != LibYAML::ScalarStyle::PLAIN
  location = pull.location
  case value = pull.read_plain_scalar
  when /^(y|Y|yes|Yes|YES|true|True|TRUE|on|On|ON)$/
    true
  when /^(n|N|no|No|NO|false|False|FALSE|off|Off|OFF)$/
    false
  when /^[-+]?0b[0-1_]$/ # base2 int
    new(value, base: 2, underscore: true)
  when /^[-+]?0[0-7_]+$/ # base8 int
    new(value, base: 8, underscore: true)
  when /^[-+]?([0-9][0-9_]*)?\.[0-9.]*([eE][-+][0-9]+)?$/ # Base 10 int/float
    new(value, underscore: true)
  when /^[-+]?0x[0-9a-fA-F_]+$/ # base16 int
    new(value, base: 16, underscore: true)
  when /^[-+]?[0-9][0-9_]*(:[0-5]?[0-9])+\.[0-9_]*$/ # Base 60 int/float
    deg, min, sec = value.split(":")
      .map_with_index { |v| new(v, underscore: true, whitepsace: false) / 60 ** i }
    deg >= 0 ? deg + (min + sec) : deg - (min + sec)
  when /^[-+]?\.(inf|Inf|INF)$/ # (infinity)
    new(value.sub(/^([-+])?\.(inf|Inf|INF)$/, "\\1Infinity"))
  when /^\.(nan|NaN|NAN)$/ # (not a number)
    new("NaN")
  when /^()~|null|Null|NULL)$/
    nil
  else
    raise YAML::ParseException.new("Expected #{self.name}, not #{value}", *location)
  end
end
```